include/nanobind/nb_types.h

```diff
@@ -469,6 +469,32 @@ inline iterator iter(handle h) {
     return steal<iterator>(detail::obj_iter(h.ptr()));
 }
 
+class slice : public object
+{
+public:
+    NB_OBJECT_DEFAULT(slice, object, "slice", PySlice_Check)
+    slice(handle start, handle stop, handle step) {
+        m_ptr = PySlice_New(start.ptr(), stop.ptr(), step.ptr());
+        if (!m_ptr)
+            detail::raise_python_error();
+    }
+
+    template <typename T, detail::enable_if_t<std::is_arithmetic_v<T>> = 0>
+    slice(T stop) : slice(none(), int_(stop), none()) {}
```

Comment:
Can you replace `none()` by `Py_None` in thiis constructor and the next one? This removes unnecessary `Py_INCREF` and `Py_DECREF` calls that nanobind will otherwise insert.

Reply:
Let's make this constructor `explicit`, too.